secboot/encrypt_tpm_test.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build !nosecboot
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot_test
+
+import (
+	"errors"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/secboot"
+)
+
+func (s *encryptSuite) TestFormatEncryptedDevice(c *C) {
+	for _, tc := range []struct {
+		initErr error
+		err     string
+	}{
+		{initErr: nil, err: ""},
+		{initErr: errors.New("some error"), err: "some error"},
+	} {
+		// create empty key to prevent blocking on lack of system entropy
+		myKey := secboot.EncryptionKey{}
```

Comment:
Given that myKey is used a lot maybe we add it to the encrptionSuite:
```
diff --git a/gadget/install/encrypt_test.go b/gadget/install/encrypt_test.go
index e54da434e1..07d92b160f 100644
--- a/gadget/install/encrypt_test.go
+++ b/gadget/install/encrypt_test.go
@@ -37,6 +37,9 @@ type encryptSuite struct {
        testutil.BaseTest
 
        mockCryptsetup *testutil.MockCmd
+
+       mockedEncryptionKey secboot.EncryptionKey
+       mockedRecoveryKey   secboot.RecoveryKey
 }
 
 var _ = Suite(&encryptSuite{})
@@ -56,6 +59,14 @@ var mockDeviceStructure = gadget.OnDiskStructure{
 func (s *encryptSuite) SetUpTest(c *C) {
        dirs.SetRootDir(c.MkDir())
        c.Assert(os.MkdirAll(dirs.SnapRunDir, 0755), IsNil)
+
+       // create empty key to prevent blocking on lack of system entropy
+       s.mockedEncryptionKey = secboot.EncryptionKey{}
+       for i := range s.mockedEncryptionKey {
+               s.mockedEncryptionKey[i] = byte(i)
+       }
+       s.mockedRecoveryKey = secboot.RecoveryKey{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0}
+
 }
```
and just use it?

Reply:
Mocked keys moved to encryption suite.